hscontrol/types/config_test.go

```diff
@@ -333,3 +334,59 @@ tls_letsencrypt_challenge_type: TLS-ALPN-01
 	err = LoadConfig(tmpDir, false)
 	assert.NoError(t, err)
 }
+
+// OK
+// server_url: headscale.com, base: clients.headscale.com
+// server_url: headscale.com, base: headscale.net
+//
+// NOT OK
+// server_url: server.headscale.com, base: headscale.com
+func TestSafeServerURL(t *testing.T) {
+	tests := []struct {
+		serverURL, baseDomain,
+		wantErr string
+	}{
+		{
```

Comment:
another case that should be allowed is if you completely ignore "real" TLDs:

serverURL: headscale.kristoffer.com
baseDomain: mybase

resulting in nodes having: `node1.mybase`, `server1.mybase`.